client/src/components/vaults/VaultTransaction/VaultTransaction.js

```diff
@@ -0,0 +1,171 @@
+        res = await bank.service.withdraw(newValue, setTx);
+      }
+
+      if (activeTransaction === "deposit") {
+        res = await bank.service.deposit(newValue, setTx);
+      }
+
+      if (activeTransaction === "borrow") {
+        res = await bank.service.borrow(newValue, setTx);
+      }
+
+      if (activeTransaction === "repay") {
+        res = await bank.service.repay(newValue, setTx);
+      }
+
+      if (res.error) {
+        setError("Transaction Error");
+        setLoading(false);
+        setTx(null);
+      } else {
+        dispatch({ type: "refreshBanks" });
+        setActiveTransaction(null);
+      }
+    }
+  };
+
+  const isCollateral =
+    activeTransaction === "withdraw" || activeTransaction === "deposit";
+
+  const handleCancel = () => {
+    setNewValue(null);
+    setActiveTransaction(null);
+  };
+
+  const handleChange = (e) => {
+    setNewValue(e.target.value);
+  };
+
+  const needsRepayUnlock = () => {
+    const isRepay = activeTransaction === "repay";
+    const needsApproval =
+      +bank.data.debtToken.unlockedAmount === 0 ||
+      +newValue > +bank.data.debtToken.unlockedAmount;
+    return isRepay && needsApproval && !localApproved;
+  };
+
+  const data = bank.data;
+  const vaultTxCalcValues = getVaultTxCalcValues(
+    data,
+    activeTransaction,
+    newValue
+  );
+
+  return (
+    <div className="VaultTransaction">
+      {loading ? (
+        <>
+          <Loading size="small" />
+          {tx ? <EtherscanLink path="tx" hash={tx} /> : null}
+        </>
+      ) : (
+        <>
+          <div className="VaultTransaction__preview">
+            <div className="VaultDetail">
+              <p>New collateralization ratio</p>
+              <h3>{vaultTxCalcValues.newCollateralizationRatio.toFixed(2)} %</h3>
+            </div>
+            <div className="VaultDetail">
+              <p>New liquidation price</p>
+              <h3>
+                {vaultTxCalcValues.newLiquidationPrice.toFixed(4)}{" "}
+                {data.collateralToken.symbol}
+                /USD
+              </h3>
+            </div>
+          </div>
+          <div className="VaultTransaction__form__Total">
+          <div className="VaultTransaction__form">
+            {needsRepayUnlock() ? (
+              <>
+                <p>Please give allowance for your repay to continue.</p>
+                <ApproveToken
+                  tokenAddress={bank.data.debtToken.address}
+                  bankAddress={bank.service.contractAddr}
+                  setError={setError}
+                  setLocalApproved={setLocalApproved}
+                />
+              </>
+            ) : (
+              <>
+                <p>
+                  How much{" "}
+                  {isCollateral
+                    ? bank.data.collateralToken.symbol
+                    : bank.data.debtToken.symbol}{" "}
+                  do you wish to {activeTransaction}?
+                </p>
+
+                <Input
```

Comment:
When repaying, the default value needs to be the max repay amount, otherwise the user will have no way to totally repay the full amount, which is 18 decimal places. The UI only shows 4 decimals (which is fine) but there needs to be a way to repay the max. Either add max repay button, or make the default value the max amount. I favor the latter option
![Screen Shot 2020-08-28 at 9 55 13 AM](https://user-images.githubusercontent.com/5077880/91573453-e7745500-e914-11ea-8ba2-ebed56118c35.png)